File: Sep03/Troy/Q6.java
class Solution {
    public int findMin(int[] nums) {
        int start = 0;
        int end = nums.length - 1;
        while(start < end) {
            int middle = (start + end) / 2;
            if (nums[middle] > nums[end]) {
                start = middle + 1;
            } else if (nums[middle] < nums[end]) {
                end = middle;
            } else if (nums[middle] == nums[end]){
                end -= 1;
            }
        }
        return nums[end];
    }
}

File: Aug14/Carl/Q6.py
class Solution(object):
    def arrayPairSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums = sorted(nums)
        totalSum = 0
        for i in range(0, len(nums), 2):
            totalSum += nums[i]
        return totalSum

a = [1,2,3,4]
print(len(a))
